wen-restart/src/wen_restart.rs

```diff
@@ -196,7 +232,100 @@ pub(crate) fn aggregate_restart_last_voted_fork_slots(
             sleep(Duration::from_millis(time_left));
         }
     }
-    Ok(())
+    Ok(last_voted_fork_slots_aggregate.get_final_result())
+}
+
+pub(crate) fn find_heaviest_fork(
+    aggregate_final_result: LastVotedForkSlotsFinalResult,
+    bank_forks: Arc<RwLock<BankForks>>,
+    exit: Arc<AtomicBool>,
+) -> Result<(Slot, Hash)> {
+    // Under normal cases without duplicate blocks, we should have a single chain of
+    // blocks from the root to the newest block with > (67% - 5% -  (100% - X%)) of stake.
+    // 1. If X% of the validators joined the restart (X > 80%), then there can't be two
+    //    kids B and C of the same parent A with more than (67% - 5% -  (100% - X%)) of
+    //    the stake each. Assume B and C exists, their combined stake is more than
+    //    2 * (67% - 5% - (100% - X%)) = 134% - 10% - 2 * (100% - X%) = 2*X% - 76% > X%.
```

Comment:
Well somebody could double vote maliciously, which means the combined stake would double count them, which would break this proof

I think the proof needs the additional constraint that the beginning split for a fork has to be at least 47-33, and cannot be something like 40-40

Reply:
Moved the whole discussion to SIMD.